addons/mrp/migrations/10.0.2.0/post-migration.py

```diff
@@ -369,36 +378,28 @@ def fill_stock_move_unit_factor(env):
         GROUP BY sm.production_id)
         as qty_done_tb
         WHERE qty_done_tb.production_id = mp.id
-        AND %(qty_produced_field)s IS NULL;
+        AND %(qty_produced_field)s IS NULL
+        AND mp.state NOT IN ('done', 'cancel')
         """ % {
         'quantity_done_field': quantity_done_field,
         'qty_produced_field': qty_produced_field,
     })
 
-    mp_obj = env['mrp.production']
-    domain = [('state', 'not in', ['done', 'cancel'])]
-    offset = 0
-    while len(mp_obj.search(
-            domain, offset=offset, limit=2000, order='id')) > 0:
-        productions = mp_obj.search(
-            domain, offset=offset, limit=2000, order='id')
-        openupgrade.logged_query(cr, """
-            UPDATE stock_move sm
-            SET unit_factor = CASE
-                WHEN (mp.product_qty - %(field)s) = 0.0
-                THEN sm.product_uom_qty
-                WHEN (mp.product_qty - %(field)s) != 0.0
-                THEN sm.product_uom_qty / (mp.product_qty - %(field)s)
-                ELSE unit_factor
-            END
-            FROM mrp_production mp
-            WHERE sm.raw_material_production_id = mp.id
-            AND sm.raw_material_production_id in %(mp_ids)s
-            """ % {
-            'field': qty_produced_field,
-            'mp_ids': tuple(productions.ids,),
-        })
-        offset += 2000
+    openupgrade.logged_query(cr, """
```

Comment:
Depending on number of data to update, the mass update can become too long than update by block of 2000 rows.

Reply:
I am re-running a migration so we will have data to compare really soon